flash/core/integrations/icevision/adapter.py

```diff
@@ -110,6 +110,27 @@ def _wrap_collate_fn(collate_fn, samples):
             DataKeys.METADATA: metadata,
         }
 
+    def change_collate_fn_dataloader(self, data_loader):
+        # Starting PL 1.7.0 - changing attributes after the DataLoader is initialized - will not work
+        # So we manually set the collate_fn to the wrapped version, for now.
+        new_kwargs = getattr(data_loader, "__pl_saved_kwargs", None)
+        new_collate_fn = functools.partial(self._wrap_collate_fn, data_loader.collate_fn)
+        if new_kwargs:
+            new_kwargs["collate_fn"] = new_collate_fn
+            setattr(data_loader, "__pl_saved_kwargs", new_kwargs)
+        data_loader.collate_fn = new_collate_fn
+        return data_loader
+
+    def update_collate_fn_dataloader(self, new_collate_fn, data_loader):
```

Comment:
Tiny nit, but maybe the should be protected?

Reply:
Fixed :)